cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -232,8 +232,7 @@ func (scaleSet *ScaleSet) waitForDeleteInstances(future *azure.Future, requiredI
 
 	defer func() {
 		if err != nil {
-			klog.Errorf("Failed to delete instances %v. Invalidating the cache to get the real scale set size", requiredIds.InstanceIds)
-			scaleSet.invalidateStatusCacheWithLock()
+			klog.Errorf("Failed to delete instances %v", requiredIds.InstanceIds)
```

Comment:
failure in the async operation is rare. VMSS updates its goal capacity when the delete request is sent and commits to that goal.  The `statusCache` refresh rate is 15 seconds so we can depend on that safely.